sysutils/apcupsd/src/opnsense/mvc/app/models/OPNsense/Apcupsd/Apcupsd.xml

```diff
@@ -0,0 +1,163 @@
+<model>
+    <mount>//OPNsense/apcupsd</mount>
+    <version>0.0.2</version>
+    <description>APC UPS configuration</description>
+    <items>
+        <!-- Container -->
+        <general>
+            <!-- field -->
+            <Enabled type="BooleanField">
+                <default>0</default>
+                <Required>Y</Required>
+            </Enabled>
+            <UPSName type="TextField">
+                <default>Firewall_UPS</default>
+                <Required>Y</Required>
+                <ValidationMessage>A name for the UPS is required.</ValidationMessage>
+            </UPSName>
+            <UPSCable type="OptionField">
+                <default>smart</default>
+                <Required>Y</Required>
+                <OptionValues>
+                    <option value="simple">Simple</option>
+                    <option value="smart">Smart</option>
+                    <option value="ether">Ether</option>
+                    <option value="usb">USB</option>
+                    <option value="940-0020B">940-0020B</option>
+                    <option value="940-0020C">940-0020C</option>
+                    <option value="940-0023A">940-0023A</option>
+                    <option value="940-0024B">940-0024B</option>
+                    <option value="940-0024C">940-0024C</option>
+                    <option value="940-0024G">940-0024G</option>
+                    <option value="940-0095A">940-0095A</option>
+                    <option value="940-0095B">940-0095B</option>
+                    <option value="940-0095C">940-0095C</option>
+                    <option value="940-0119A">940-0119A</option>
+                    <option value="940-0127A">940-0127A</option>
+                    <option value="940-0128A">940-0128A</option>
+                    <option value="940-0625A">940-0625A</option>
+                    <option value="940-1524C">940-1524C</option>
+                    <option value="MAM-04-02-2000">MAM-04-02-2000</option>
+                </OptionValues>
+            </UPSCable>
+            <UPSType type="OptionField">
+                <default>apcsmart</default>
+                <Required>Y</Required>
+                <OptionValues>
+                    <apcsmart>apcsmart</apcsmart>
+                    <usb>usb</usb>
+                    <net>net</net>
+                    <snmp>snmp</snmp>
+                    <netsnmp>netsnmp</netsnmp>
+                    <dumb>dumb</dumb>
+                    <pcnet>pcnet</pcnet>
+                    <modbus>modbus</modbus>
+                </OptionValues>
+            </UPSType>
+            <Device type="TextField">
+                <Required>N</Required>
+                <default>/dev/cuaU0</default>
+            </Device>
+            <Polltime type="IntegerField">
+                <Required>N</Required>
```

Comment:
If you don't use if not != '' in templating I would make this and other fields as required. Otherwise these fields are empty in config and may overwrite defaults of the software itself. Sorry, I never used apcupsd, so only guessing.

Reply:
May need to look into this and get back. 👍 
//Dan Lundqvist (mrzaz)

Reply:
Could you Nicola [xbb](https://github.com/xbb) look into this with [mimugmail](https://github.com/mimugmail) so this gets right ?
//Danne (mrzaz)

Reply:
> If you don't use if not != '' in templating I would make this and other fields as required. Otherwise these fields are empty in config and may overwrite defaults of the software itself. Sorry, I never used apcupsd, so only guessing.

Yes POLLTIME should be required, by default it is 60 and it's commented out in the sample config. 

It still works fine when empty because there is `default("60")` also in the template file

I'll take a look at all the variables and check with the sample config.

Should I remove all the `default(...)` from the template and have them only in the model?